snapcraft/tests/test_project_loader.py

```diff
@@ -885,6 +885,30 @@ def test_invalid_environment(self, mock_load_plugin):
             "The 'environment/INVALID' property does not match the required "
             "schema: \[1, 2\].*")
 
+    @unittest.mock.patch('snapcraft.internal.parts.PartsConfig.load_plugin')
+    def test_invalid_yaml_empty_version(self, mock_loadPlugin):
+        fake_logger = fixtures.FakeLogger(level=logging.ERROR)
+        self.useFixture(fake_logger)
+
+        self.make_snapcraft_yaml("""name: test
+version:''
+summary: test
+description: test
+confinement: strict
+grade: stable
+parts:
+  part1:
+    plugin: go
+    stage-packages: [fswebcam]
+""")
+        raised = self.assertRaises(
+            errors.SnapcraftSchemaError,
+            project_loader.Config)
+
+        self.assertEqual(
+            raised.message,
+            "'version' can not be an empty string")
```

Comment:
I suspect this will fail because we are showing the regexp in the error.
We want to change that in the future to something nicer, but for now, it's ok just to update the expected value here.